File: app/src/main/java/com/tunasteak/activity/RecyclerViewBindingActivity.java
package com.tunasteak.activity;

import android.view.View;

import com.tunasteak.R;
import com.tunasteak.databinding.ActivityRecyclerViewBinding;
import com.tunasteak.model.RecyclerViewModel;
import com.tunasteak.model.TBindingModel;

public class RecyclerViewBindingActivity extends TBindingActivity<ActivityRecyclerViewBinding> {
    @Override
    protected int getLayoutId() {
        return R.layout.activity_recycler_view;
    }

    @Override
    protected TBindingModel getModel() {
        return new RecyclerViewModel(binding, getIntent());
    }

    @Override
    public void onInit() {

    }

    @Override
    public void onClick(View v) {
    }
}

File: app/src/main/java/com/tunasteak/activity/RecyclerViewBindingActivity.kt
package com.tunasteak.activity

import android.view.View
import com.tunasteak.demo.R
import com.tunasteak.demo.databinding.ActivityRecyclerViewBinding
import com.tunasteak.model.TBindingModel
import com.tunasteak.model.RecyclerViewModel

class RecyclerViewBindingActivity : TBindingActivity<ActivityRecyclerViewBinding?>() {
    override fun getLayoutId(): Int {
        return R.layout.activity_recycler_view
    }

    override fun getModel(): TBindingModel<*> {
        return RecyclerViewModel(binding, intent)
    }

    override fun getData(): Any? {
        return null
    }

    override fun onInit() {}
    override fun onClick(v: View) {}
}